CONTRIBUTING.md

```diff
@@ -1,24 +1,63 @@
 # Contributing
 
-Lumigator is still in early stages of development and as such will have
-large variance between even PATCH versions in terms of feature development.
+Lumigator is still in early stages of development and as such will have large variance between even PATCH versions in terms of feature development. To stay updated on the latest announcements, join our Discord channel: [Discord Channel](https://discord.com/channels/1089876418936180786/1281660143251095634).
 
-We want to encourage both suggestions and contributions from the community in the near future. Before submitting a PR, please open a discussion in a related issue.
+We want to encourage both suggestions and contributions from the community in the near future. Before submitting a [pull request](https://github.com/mozilla-ai/lumigator/pulls), please open a discussion in a [related issue](https://github.com/mozilla-ai/lumigator/issues).
 
-The lumigator project contains several packages: the `lumigator/python/mzai/backend` package that powers the lumigator server functionality, the `lumigator/python/mzai/schemas` package containing the formal schemas for communication with the server through the REST API, and the `lumigator/python/mzai/sdk` package abstracting the REST API for Python applications. Each package holds its own `pyproject.toml` definition.
+**Before you contribute**
+
+* Check out the roadmap: This will give you an overview of the project's direction and help you identify areas where your contributions can have the greatest impact.
+* Take a look into the existing issues: Check if your issue or challenge has already been discussed or reported. This helps avoid duplication of effort.
+* Learn about the code: The Lumigator project consists of several packages, each with its own pyproject.toml file. Understanding the project structure will make it easier to contribute effectively. The lumigator project contains several packages: the `lumigator/python/mzai/backend` package that powers the lumigator server functionality, the `lumigator/python/mzai/schemas` package containing the formal schemas for communication with the server through the REST API, and the `lumigator/python/mzai/sdk` package abstracting the REST API for Python applications. Each package holds its own `pyproject.toml` definition.
```

Comment:
```suggestion
* Learn about the code: The Lumigator project consists of several packages, each with its own pyproject.toml file. Understanding the project structure will make it easier to contribute effectively: the `lumigator/python/mzai/backend` package that powers the lumigator server functionality, the `lumigator/python/mzai/schemas` package containing the formal schemas for communication with the server through the REST API, and the `lumigator/python/mzai/sdk` package abstracting the REST API for Python applications. Each package holds its own `pyproject.toml` definition.
```

Reply:
Do we also want to mention the `frontend` and `infra` here? Should this be an exhaustive list, or a few examples of where's what?